public-api/src/main/java/org/pmiops/workbench/service/AchillesAnalysisService.java

```diff
@@ -0,0 +1,742 @@
+            for (int k=0; k<maleBinRanges.size()-1;k++) {
+                maleRangesInResults.add((trimTrailingZeroDecimals(String.valueOf(maleBinRanges.get(k))) + " - " + trimTrailingZeroDecimals(String.valueOf(maleBinRanges.get(k+1)))));
+            }
+            maleRangesInResults.add((">= " + trimTrailingZeroDecimals(String.valueOf(maleBinRanges.get(maleBinRanges.size()-1)))));
+
+            ArrayList<String> femaleRangesInResults = new ArrayList<>();
+            femaleRangesInResults.add(("< " + trimTrailingZeroDecimals(String.valueOf(femaleBinRanges.get(0)))));
+            for (int k=0; k<femaleBinRanges.size()-1;k++) {
+                femaleRangesInResults.add((trimTrailingZeroDecimals(String.valueOf(femaleBinRanges.get(k))) + " - " + trimTrailingZeroDecimals(String.valueOf(femaleBinRanges.get(k+1)))));
+            }
+            femaleRangesInResults.add((">= " + trimTrailingZeroDecimals(String.valueOf(femaleBinRanges.get(femaleBinRanges.size()-1)))));
+
+            ArrayList<String> otherRangesInResults = new ArrayList<>();
+            otherRangesInResults.add(("< " + trimTrailingZeroDecimals(String.valueOf(otherBinRanges.get(0)))));
+            for (int k=0; k<otherBinRanges.size()-1;k++) {
+                otherRangesInResults.add((trimTrailingZeroDecimals(String.valueOf(otherBinRanges.get(k))) + " - " + trimTrailingZeroDecimals(String.valueOf(otherBinRanges.get(k+1)))));
+            }
+            otherRangesInResults.add((">= " + trimTrailingZeroDecimals(String.valueOf(otherBinRanges.get(otherBinRanges.size()-1)))));
+
+            for(AchillesResult ar: aa.getResults()){
+                String analysisStratumName=ar.getAnalysisStratumName();
+                String result_value = ar.getStratum4();
+
+                if(Long.valueOf(ar.getStratum3()) == MALE && maleRangesInResults.contains(result_value)){
+                    maleRangesInResults.remove(result_value);
+                }else if(Long.valueOf(ar.getStratum3()) == FEMALE && femaleRangesInResults.contains(result_value)){
+                    femaleRangesInResults.remove(result_value);
+                }else if(Long.valueOf(ar.getStratum3()) == OTHER && otherRangesInResults.contains(result_value)){
+                    otherRangesInResults.remove(result_value);
+                }
+
+                if (analysisStratumName == null || analysisStratumName.equals("")) {
+                    ar.setAnalysisStratumName(genderStratumNameMap.get(ar.getStratum3()));
+                }
+            }
+
+            for(String maleRemaining: maleRangesInResults){
+                String missingBinWidth = null;
+                if (maleBinWidth == (long)(maleBinWidth)) {
+                    missingBinWidth = String.format("%d",(long)maleBinWidth);
+                } else {
+                    missingBinWidth = String.format("%.2f", maleBinWidth);
+                }
+                missingBinWidth = trimTrailingZeroDecimals(missingBinWidth);
```

Comment:
missingBinWidth isn't referenced anywhere.

Reply:
Changed this thanks